libs/sdk-ui-kit/src/@ui/UiMenu/UiMenu.tsx

```diff
@@ -53,34 +79,50 @@ export function UiMenu<T extends IUiMenuItemData = object>(props: UiMenuProps<T>
         [items, focusedId],
     );
 
+    React.useEffect(() => {
+        // Only focus when shownCustomContentItemId becomes undefined (was previously set)
+        if (shownCustomContentItemId === undefined && menuComponentRef.current) {
+            menuComponentRef.current.focus();
+        }
+    }, [shownCustomContentItemId, menuComponentRef]);
+
+    const menuClassName = typeof className === "function" ? className(contextStoreValue) : className;
+
     return (
         <UiMenuContextStore value={contextStoreValue}>
             <div
-                className={cx(b(), b({ controlType }), className)}
+                className={cx(b(), b({ controlType }), menuClassName)}
                 style={{ maxWidth }}
                 onKeyDownCapture={() => setControlType("keyboard")}
                 onMouseMoveCapture={() => setControlType("mouse")}
             >
-                <MenuHeaderComponent />
-
-                <div
-                    className={e("items-container")}
-                    ref={itemsContainerRef as React.MutableRefObject<HTMLDivElement>}
-                >
-                    <menu
-                        className={e("items")}
-                        tabIndex={0}
-                        onKeyDown={handleKeyDown}
-                        aria-activedescendant={focusedItem ? makeItemId(focusedItem) : undefined}
-                        {...ariaAttributes}
-                        role="menu"
-                        ref={menuComponentRef}
-                    >
-                        {currentMenuLevelItems.map((item, index) => (
-                            <ItemComponent key={"id" in item ? item.id : index} item={item} />
-                        ))}
-                    </menu>
-                </div>
+                {shownCustomContentItemId ? (
+                    <ContentWrapper keyboardNavigationHandler={handleKeyDownInCustomContent}>
+                        <Content item={getContentItem(items, shownCustomContentItemId)} />
+                    </ContentWrapper>
+                ) : (
+                    <>
+                        <MenuHeader />
+                        <div
+                            className={e("items-container")}
+                            ref={itemsContainerRef as React.MutableRefObject<HTMLDivElement>}
+                        >
+                            <menu
+                                className={e("items")}
+                                tabIndex={0}
+                                onKeyDown={handleKeyDown}
+                                aria-activedescendant={focusedItem ? makeItemId(focusedItem) : undefined}
```

Comment:
Maybe this can be just `makeItemId(focusedItem)` and `undefined` logic will be inside of the function?

Reply:
fixed